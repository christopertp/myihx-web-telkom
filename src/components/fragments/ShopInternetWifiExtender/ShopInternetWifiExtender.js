import React, { useState } from 'react';
import { useHistory } from 'react-router-dom';
import styles from './styles.scoped.css';
import Bullet from '../../elements/Bullet/Bullet';
import Button from '../../elements/Button';
import ModalLogin from '../../elements/ModalLogin';
import Spinner from '../../elements/Spinner';
import StickyPrice from '../../elements/StickyPrice';
import { getUserData } from '../../../utils/storage';

export default function ShopInternetWifiExtender() {
  const [modalLogin, setModalLogin] = useState(false);
  const isLoggedIn = !!getUserData();
  const history = useHistory();

  const featuresData = [
    { title: 'Unlimited high speed internet', subtitle: 'Unlimited high speed internet' },
    { title: 'Automatic seamless connection', subtitle: 'Connect autimatically when your device is in range of a hotspot' },
    { title: 'Connect up to 5 devices', subtitle: 'Use the same username for up to 5 devices' }
  ];

  const activateData = [
    { title: 'Datang ke Plasa Telkom terdekat', subtitle: 'Lorem ipsum dolor sit amet 1' },
    { title: 'Daftar dengan nomor IndiHome Anda', subtitle: 'Lorem ipsum dolor sit amet 2' },
    { title: 'Teknisi kami akan datang untuk memasang wifi extender Anda', subtitle: 'Lorem ipsum dolor sit amet 3' }
  ];

  const isLoading = {
    data: false,
    submit: false
  };

  const onClickSubscribe = () => {
    isLoggedIn ? history.push('/order/wifi-extender') : setModalLogin(true);
  };

  const onClickDetail = () => {};

  return (
    <>
      <section className={styles.banner}>
        <h2>Complete coverage with wifi extender</h2>
        <p>Use wifi extender for a stronger internet connection in your house/office</p>
        <img alt="wifi extender" src="/assets/bnr_internet_wifiextender.png" />
      </section>
      <section className={styles.about}>
        <h5>About this product</h5>
        <p>
          Mau internetan dengan jaringan wifi yang lebih luas di rumah Anda? IndiHome kini
          menyediakan perangkat Wifi Extender untuk memperkuat pemancaran sinyal wifi di setiap area
          sudut rumah Anda tanpa harus menyediakan kabel tambahan.
        </p>
        <p>Dengan Wifi Extender dari
          IndiHome, internetan di setiap sudut rumah jadi semakin leluasa.
          Hanya <b>Rp35.000/bulan</b>.
        </p>
      </section>
      <section className={styles.grid}>
        <section className={styles.features}>
          <h5>Features</h5>
          <ul>
            {featuresData.map((i, idx) =>
              (<li key={idx}>
                <img alt="hybrid box" src="/assets/grfx_line_hybridbox_1.png" />
                <h5>
                  {i.title}
                </h5>
                <p>{i.subtitle}</p>
              </li>)
            )}
          </ul>
        </section>
        <section className={styles.activate}>
          <h5>How to activate</h5>
          <ul>
            {activateData.map((i, idx) =>
              (<li key={idx}>
                <Bullet connect={idx < activateData.length - 1} />
                <h5>{i.title}</h5>
                <p>{i.subtitle}</p>
              </li>)
            )}
          </ul>
          <Button onClick={onClickSubscribe}>
            {isLoading.submit ? <Spinner /> : 'Activate'}
          </Button>
        </section>
      </section>
      <StickyPrice
        isLoading={isLoading}
        onClickDetail={onClickDetail}
        onClickSubscribe={onClickSubscribe}
        price={35000}
      />
      <ModalLogin onClose={() => setModalLogin(false)} open={modalLogin} />
    </>
  );
}
