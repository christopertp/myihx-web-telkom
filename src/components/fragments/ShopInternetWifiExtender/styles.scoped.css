.banner {
  background-color: #f2f2f2;
  display: grid;
  grid-template-columns: 1fr auto;
  grid-template-rows: auto auto;
  margin: -3rem var(--neg-side-margin) 0;
  padding: 7rem var(--side-margin);
}

.banner > p { 
  color: #999999;
  letter-spacing: 0.0625rem;
  margin-bottom: auto;
}

.banner > h2 {
  grid-column: 1;
  font-size: 1.75rem;
  font-weight: 800;
  line-height: 1.2;
  margin: auto 0 1rem;
}

.banner > img {
  grid-row: 1/3;
  grid-column: 2;
  height: 12.75rem;
  margin-left: 4rem;
  border-radius: 0.5rem;
}

.about {
  background-color: #ffffff;
  margin: 0 var(--neg-side-margin);
  padding: 4rem var(--side-margin) 1rem;
}

.about > h5,
.features > h5, 
.activate > h5 {
  color: #66686a;
  font-size: 0.9rem;
  font-weight: 400;
  margin-bottom: 1rem;
  text-transform: uppercase;
  letter-spacing: 0.0625rem
}

.about > p {
  color: #191a1a;
  line-height: 1.6;
  margin-bottom: 1rem;
}

.grid {
  background-color: #ffffff;
  margin: 0 var(--neg-side-margin);
  padding: 2rem var(--side-margin) 2rem;
  display: flex;
}

.grid > section {
  flex: 1;
}

.features {
  margin-right: 1rem;
}

.features > ul > li {
  display: grid;
  grid-template-columns: fit-content(100%) auto;
  grid-template-rows: auto auto;
  max-width: 36rem;
  padding: 0.5rem 0rem;
}
.features > ul > li:first-child {
  padding-top: 0;
}
.features > ul > li > img {
  grid-row: 1/3;
  width: 6rem;
  margin-right: 1rem;
} 
.features > ul > li > h5 {
  font-size: 1.2em;
  margin: auto 0 0.3rem;
}

.activate > ul {
  margin-top: 2rem;
}
.activate > ul > li {
  display: grid;
  grid-template-columns: 2rem auto;
  grid-template-rows: auto auto;
  max-width: 36rem;
}
.activate > ul > li > div {
  grid-row: 1/3;
  margin-right: auto;
  padding-top: 0.5rem;
}
.activate > ul > li > h5 {
  font-size: 1.2em;
  margin: auto 0 0.3rem;
}
.activate > ul > li > p {
  padding-bottom: 1.75rem;
}

@media only screen and (max-width: 48em) {
  .banner {
    grid-template-columns: 2fr minmax(0, 1fr);
    background: url(/assets/bnr_internet_wifiextender.png) center center no-repeat;
    background-size: cover;
    margin: 0rem -1rem 0;
    padding: 1.5rem 1.25rem;
  }
  .banner > h2 {
    color: #ffffff;
    font-size: 1.25rem;
    margin: 0 0 0.5rem;
  }
  .banner > p {
    color: #ffffff;
    font-size: 0.65rem;
    margin-bottom: 0;
    grid-row: 2;
  }
  .banner > img {
    display: none;
  }
  .features > ul {
    grid-template-columns: auto;
  }
  .about {
    padding-top: 2rem;
  }
  .about > p {
    font-size: 0.8rem;
  }
  .grid {
    flex-direction: column;
  }
  .features {
    padding-top: 0;
    padding-bottom: 1rem;
  }
  .features > ul > li > img {
    width: 3.75rem;
  }
  .features > ul > li > h5,
  .activate > ul > li > h5 {
    font-size: 0.8rem;
  }
  .features > ul > li > p,
  .activate > ul > li > p {
    font-size: 0.75rem;
  }
  .activate > ul {
    margin-top: 0;
  }
  .activate > button {
    display: none;
  }
}
