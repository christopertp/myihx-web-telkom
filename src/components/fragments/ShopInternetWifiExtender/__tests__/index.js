import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import ShopInternetWifiExtender from '../ShopInternetWifiExtender';

describe('src/components/fragments/ShopInternetWifiExtender', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<ShopInternetWifiExtender />);
    expect(tree).toMatchSnapshot();
  });
});
