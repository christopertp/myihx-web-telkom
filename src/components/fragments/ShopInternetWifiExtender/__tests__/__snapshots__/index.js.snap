// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/fragments/ShopInternetWifiExtender render 1`] = `
<React.Fragment>
  <section>
    <h2>
      Complete coverage with wifi extender
    </h2>
    <p>
      Use wifi extender for a stronger internet connection in your house/office
    </p>
    <img
      alt="wifi extender"
      src="/assets/bnr_internet_wifiextender.png"
    />
  </section>
  <section>
    <h5>
      About this product
    </h5>
    <p>
      Mau internetan dengan jaringan wifi yang lebih luas di rumah Anda? IndiHome kini menyediakan perangkat Wifi Extender untuk memperkuat pemancaran sinyal wifi di setiap area sudut rumah Anda tanpa harus menyediakan kabel tambahan.
    </p>
    <p>
      Dengan Wifi Extender dari IndiHome, internetan di setiap sudut rumah jadi semakin leluasa. Hanya 
      <b>
        Rp35.000/bulan
      </b>
      .
    </p>
  </section>
  <section>
    <section>
      <h5>
        Features
      </h5>
      <ul>
        <li>
          <img
            alt="hybrid box"
            src="/assets/grfx_line_hybridbox_1.png"
          />
          <h5>
            Unlimited high speed internet
          </h5>
          <p>
            Unlimited high speed internet
          </p>
        </li>
        <li>
          <img
            alt="hybrid box"
            src="/assets/grfx_line_hybridbox_1.png"
          />
          <h5>
            Automatic seamless connection
          </h5>
          <p>
            Connect autimatically when your device is in range of a hotspot
          </p>
        </li>
        <li>
          <img
            alt="hybrid box"
            src="/assets/grfx_line_hybridbox_1.png"
          />
          <h5>
            Connect up to 5 devices
          </h5>
          <p>
            Use the same username for up to 5 devices
          </p>
        </li>
      </ul>
    </section>
    <section>
      <h5>
        How to activate
      </h5>
      <ul>
        <li>
          <Bullet
            className=""
            color={
              Array [
                "#ee3124",
                "#cccccc",
              ]
            }
            connect={true}
            size="0.75rem"
          />
          <h5>
            Datang ke Plasa Telkom terdekat
          </h5>
          <p>
            Lorem ipsum dolor sit amet 1
          </p>
        </li>
        <li>
          <Bullet
            className=""
            color={
              Array [
                "#ee3124",
                "#cccccc",
              ]
            }
            connect={true}
            size="0.75rem"
          />
          <h5>
            Daftar dengan nomor IndiHome Anda
          </h5>
          <p>
            Lorem ipsum dolor sit amet 2
          </p>
        </li>
        <li>
          <Bullet
            className=""
            color={
              Array [
                "#ee3124",
                "#cccccc",
              ]
            }
            connect={false}
            size="0.75rem"
          />
          <h5>
            Teknisi kami akan datang untuk memasang wifi extender Anda
          </h5>
          <p>
            Lorem ipsum dolor sit amet 3
          </p>
        </li>
      </ul>
      <Button
        className=""
        disabled={false}
        onClick={[Function]}
        to=""
        type="button"
        variant="red"
      >
        Activate
      </Button>
    </section>
  </section>
  <StickyPrice
    appearOn={17.25}
    className=""
    isLoading={
      Object {
        "data": false,
        "submit": false,
      }
    }
    onClickDetail={[Function]}
    onClickSubscribe={[Function]}
    price={35000}
  />
  <ModalLogin
    onClose={[Function]}
    open={false}
  />
</React.Fragment>
`;
